tests.c

```diff
@@ -21,15 +21,66 @@ clean_suite(void)
 
 /* Add tests here */
 void
-test_func(void)
+test_opcode_0x01(void) // NOLINT
 {
-  ;
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu_write_mem(&cpu, 0x0001, 0xFF); // NOLINT
+  cpu_write_mem(&cpu, 0x0002, 0x12); // NOLINT
+
+  int code_found = execute_instruction(&cpu, 0x01); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT(cpu.pc == 3);
+  CU_ASSERT(cpu.c == 0xFF);
+  CU_ASSERT(cpu.b == 0x12);
+
+  // clean up
+  cpu_write_mem(&cpu, 0x0001, 0x00);
+  cpu_write_mem(&cpu, 0x0002, 0x00);
 }
 
 void
-test_func2(void)
+test_opcode_0x0d(void) // NOLINT
 {
-  ;
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.c = 0x02;
+
+  // decrease C from 2 to 1
+  int code_found = execute_instruction(&cpu, 0x0d); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT(cpu.pc == 1);
+  CU_ASSERT(cpu.c == 0x01);
+  CU_ASSERT((cpu.flags & FLAG_P) == 0);
+  CU_ASSERT((cpu.flags & FLAG_S) == 0);
+  CU_ASSERT((cpu.flags & FLAG_Z) == 0);
+  CU_ASSERT((cpu.flags & FLAG_AC) == FLAG_AC);
+
+  // decrease C from 1 to 0
```

Comment:
I think we can get away with it here, but it does seem a little dangerous to include these test cases all in one unit test, especially because we don't a function that resets the cpu state before each series of asserts.